pkg/types/baremetal/validation/platform_test.go

```diff
@@ -191,10 +209,84 @@ func TestValidatePlatform(t *testing.T) {
 				BootstrapProvisioningIP: "192.168.111.5",
 				ExternalBridge:          "br0",
 				ProvisioningBridge:      "br1",
+				BootstrapOSImage:        "http://192.168.111.1/images/qemu.x86_64.qcow2.gz?sha256=3b5a882c2af3e19d515b961855d144f293cab30190c2bdedd661af31a1fc4e2f",
+				ClusterOSImage:          "http://192.168.111.1/images/metal.x86_64.qcow2.gz?sha256=340dfa4d92450f2eee852ed1e2d02e3138cc68d824827ef9cf0a40a7ea2f93da",
 			},
 			network:  network,
 			expected: "Invalid value: \"192.168.111.5\": the IP must not be in 192.168.111.0/24 subnet",
 		},
+		{
+			name: "invalid_bootstraposimage",
+			platform: &baremetal.Platform{
+				APIVIP:                  "192.168.111.2",
+				DNSVIP:                  "192.168.111.3",
+				IngressVIP:              "192.168.111.4",
+				Hosts:                   []*baremetal.Host{},
+				LibvirtURI:              "qemu://system",
+				ClusterProvisioningIP:   "172.22.0.3",
+				BootstrapProvisioningIP: "172.22.0.2",
+				ExternalBridge:          "br0",
+				ProvisioningBridge:      "br1",
+				BootstrapOSImage:        "192.168.111.1/images/qemu.x86_64.qcow2.gz?sha256=3b5a882c2af3e19d515b961855d144f293cab30190c2bdedd661af31a1fc4e2f",
+				ClusterOSImage:          "http://192.168.111.1/images/metal.x86_64.qcow2.gz?sha256=340dfa4d92450f2eee852ed1e2d02e3138cc68d824827ef9cf0a40a7ea2f93da",
+			},
+			network:  network,
+			expected: "The URI provided is invalid",
```

Comment:
would expect the test expected to be more clearer.. this doesn't ensure that the error was with `BootstrapOSImage` field..